src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/ReadOnlyListExtensions.cs

```diff
@@ -32,4 +32,47 @@ static ImmutableArray<TResult> BuildResult(IReadOnlyList<T> items, Func<T, TResu
             return results.DrainToImmutable();
         }
     }
+
+    public static Enumerable<T> AsEnumerable<T>(this IReadOnlyList<T> source) => new(source);
+
+    public readonly struct Enumerable<T>(IReadOnlyList<T> source) : IEnumerable<T>
```

Comment:
Should this be a ref struct or NonCopyable to avoid boxing it?

Reply:
Good suggestion. I think these can all be ref structs and don't need to implement `IEnumerable<T>` or `IEnumerator<T>`.